crates/ruff_linter/src/rules/ruff/rules/class_dict_annotations.rs

```diff
@@ -0,0 +1,131 @@
+use crate::checkers::ast::Checker;
+use ruff_diagnostics::{Diagnostic, FixAvailability, Violation};
+use ruff_macros::{ViolationMetadata, derive_message_formats};
+use ruff_python_ast::{Expr, ExprCall, PythonVersion};
+use ruff_text_size::Ranged;
+
+/// ## What it does
+/// Checks for uses of `<identifier>.__dict__.get("__annotations__" [, <default>])`
+/// on Python 3.10+ and Python < 3.10 with `typing_extensions` enabled.
+///
+/// ## Why is this bad?
+/// On Python 3.10 and newer, `<identifier>.__dict__.get("__annotations__" [, <default>])`
```

Comment:
This is incorrect, the `__dict__.get` approach is reliable on all Python versions up to 3.14. (On 3.14 it will only return any annotations if the class is defined under `from __future__ import annotations`.) However, it's better to use `inspect.get_annotations` or one the other functions, because (a) it avoids accessing undocumented internals and (b) it's forward-compatible with Python 3.14.

Reply:
This has been updated.